Antenna_Performance_Metric/count_jobs.py

```diff
@@ -0,0 +1,34 @@
+# Count the number of jobs from string of format:
+# 28178971_1 28178971_2 28178971_[3-50]
+import argparse
+import re
+
+def expand_array_job(job_id):
+    '''expand 28178971_[3-50] to 28178971_3, 28178971_4, ..., 28178971_50'''
+    if re.match(r'\d+_\[\d+-\d+\]$', job_id):
+        prefix, range_str = job_id.split('_')
+        start, end = map(int, range_str[1:-1].split('-'))
+        return [f"{prefix}_{i}" for i in range(start, end + 1)]
+    else:
+        return [job_id]
```

Comment:
Since you are already using a regex, you might as well use it to get the prefix/start/end too:
```suggestion
    if (m := re.match(r'(\d+)_\[(\d+)-(\d+)\]$', job_id)):
        prefix, start, end = m.groups()
        return [f"{prefix}_{i}" for i in range(int(start), int(end)+1)]
    else:
        return [job_id]
```
If you are using a Python older than 3.8 that doesn't support `:=` you can split the if into two lines:

```suggestion
    match = re.match(r'(\d+)_\[(\d+)-(\d+)\]$', job_id)
    if match:
        prefix, start, end = match.groups()
        return [f"{prefix}_{i}" for i in range(int(start), int(end)+1)]
    else:
        return [job_id]
```